substrate/client/transaction-pool/src/api.rs

```diff
@@ -236,12 +239,31 @@ where
 
 		use sp_api::Core;
 
+		// Get the priority from the `TransactionPriorityModule`.
+		// If the transaction is not listed in the module, use the priority obtained from the runtime.
+		let priority_modifier = client.get_priority(uxt.clone());
+
 		sp_tracing::within_span!(
 			sp_tracing::Level::TRACE, "runtime::validate_transaction";
 		{
 			if api_version >= 3 {
-				runtime_api.validate_transaction(at, source, uxt, at)
-					.map_err(|e| Error::RuntimeApi(e.to_string()))
+				let runtime_priority = runtime_api.validate_transaction(at, source, uxt.clone(), at)
+					.map_err(|e| Error::RuntimeApi(e.to_string()))?;
+
+				if let Ok(mut validity) = runtime_priority.clone() {
+					if let Some(new_priority) = priority_modifier {
+						// override the runtime priority by the priority from the `TransactionPriorityModule`
+						validity.priority = new_priority;
+
+						log::debug!(target: "tx_priority_module", "Transaction priority of {:?} has been overwritten to {:?}.", uxt, new_priority);
+
+						return Ok(Ok(validity))
+					}
+				}
```

Comment:
this validation is here 3x the same. 

i would suggest to do it only once after, since each path returns the same type - you should be abl to do it just after this if statements.